packages/devtools_app/lib/src/shared/ui/filter.dart

```diff
@@ -463,3 +465,156 @@ extension PatternListExtension on List<Pattern> {
         : cast<String>();
   }
 }
+
+class FilterField<T> extends StatefulWidget {
+  const FilterField({
+    super.key,
+    required this.controller,
+  });
+
+  final FilterControllerMixin<T> controller;
+
+  @override
+  State<FilterField<T>> createState() => _FilterFieldState<T>();
+}
+
+class _FilterFieldState<T> extends State<FilterField<T>> with AutoDisposeMixin {
+  late final TextEditingController queryTextFieldController;
+  late bool useRegExp;
+
+  @override
+  void initState() {
+    super.initState();
+    queryTextFieldController = TextEditingController(
+      text: widget.controller.activeFilter.value.queryFilter.query,
+    );
+    useRegExp = widget.controller.useRegExp.value;
+    addAutoDisposeListener(widget.controller.useRegExp, () {
+      useRegExp = widget.controller.useRegExp.value;
+    });
+  }
+
+  @override
+  void dispose() {
+    queryTextFieldController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: DevToolsClearableTextField(
+            autofocus: true,
+            labelText: 'Filter',
+            controller: queryTextFieldController,
+            additionalSuffixActions: [
+              DevToolsToggleButton(
+                icon: Codicons.regex,
+                message: 'Use regular expressions',
+                outlined: false,
+                isSelected: useRegExp,
+                onPressed: () => setState(() {
+                  widget.controller.useRegExp.value = !useRegExp;
+                  widget.controller.setActiveFilter(
+                    query: queryTextFieldController.value.text,
+                    toggleFilters: widget.controller._toggleFilters,
+                  );
+                }),
+              ),
+            ],
+            onChanged: (_) {
+              widget.controller.setActiveFilter(
+                query: queryTextFieldController.value.text,
+                toggleFilters: widget.controller._toggleFilters,
+              );
+            },
+          ),
+        ),
+        const SizedBox(width: defaultSpacing),
+        DevToolsFilterButton(
+          message: 'Filter Settings',
+          onPressed: () {
+            unawaited(
+              showDialog(
+                context: context,
+                builder: (context) => FilterOptionsDialog(
+                  controller: widget.controller,
+                ),
+              ),
+            );
+          },
+          isFilterActive: false,
+        ),
+      ],
+    );
+  }
+}
+
+class FilterOptionsDialog<T> extends StatefulWidget {
```

Comment:
we already have a class FilterDialog. Why are we creating a new one?

Reply:
Ah thanks for reminding me I need to rename this.
The filtering field used to be inside this dialog, the new implementation offers a field that has a button to open this dialog, which contains the toggle-able filters.